doc/topics/slots/index.rst

```diff
@@ -68,3 +68,109 @@ Here is an example of result parsing and appending:
       file.copy:
         - name: __slot__:salt:user.info(someuser).home ~ /subdirectory
         - source: salt://somefile
+
+Example Usage
+-------------
+
+In Salt, slots are a powerful feature that allows you to populate information
+dynamically within your Salt states. One of the best use cases for slots is when
+you need to reference data that is created or modified during the course of a
+Salt run.
+
+Consider the following example, where we aim to add a user named 'foobar' to a
+group named 'known_users' with specific user and group IDs. To achieve this, we
+utilize slots to retrieve the group ID of 'known_users' as it is created or
+modified during the Salt run.
+
+.. code-block:: yaml
+
+    add_group_known_users:
+      group.present:
+        - name: known_users
+
+    add_user:
+      user.present:
+        - name: foobar
+        - uid: 600
+        - gid: __slot__:salt:group.info("known_users").gid
```

Comment:
This will not work because the argument is quoted.

Reply:
works just fine.

Reply:
Hmm. Every time I've tried using quotes with slots it doesn't work properly.